server/routes/login.ts

```diff
@@ -34,34 +34,20 @@ export default function loginRouter(userAuthenticationService: UserAuthenticatio
   const notifySmsTemplate = smsTemplateId || ''
   const notifyEmailTemplate = emailTemplateId || ''
 
-  const postLogin = async (req: Request, res: Response) => {
-    // break out for auth users - shouldn't have got to this page anyway
-    if (req.hmppsAuthMFAUser) {
+  const postLogin = async (req: Request, res: Response, next: NextFunction) => {
+    const userAuthenticationDetails = await userAuthenticationService.getUserAuthenticationDetails(req.user.username)
+
+    // break out for auth users
+    if (req.hmppsAuthMFAUser || userAuthenticationDetails.length === 0) {
       res.redirect(`/calendar/${utilities.getStartMonth()}#today`)
       return
     }
 
     const ipAddress = req.get('x-forwarded-for') || req.socket.remoteAddress || ''
 
-    let userNotSignedUpMessage = false
-
     try {
-      const userAuthenticationDetails = await userAuthenticationService.getUserAuthenticationDetails(req.user.username)
-
-      if (userAuthenticationDetails === null || userAuthenticationDetails.length === 0) {
-        processError(
-          new Error(`Error : No Sms or Email address returned for QuantumId : ${req.user.username}`),
-          req,
-          res,
-          true,
-        )
-        return
-      }
-
```

Comment:
When no userAuthenticationDetails in DB, we now break out as for users with the migrate role